### DISCUSSION TITLE
Kit stops working after some time

### DISCUSSION THREAD

**Discussion by jacopo-degattis** [johnlindquist/kit#912] [Errors]
Hi everyone,
Recently I'm struggling to work with Kit.
When I write to scripts (using typescript mode) and I try to execute them it gives me the following error:
```
Cannot find module '/Users/{user}/.kenv/.scripts/testt.mjs' imported from /Users/{user}/.kit/api/kit.js
Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/Users/{user}/.kenv/.scripts/testt.mjs' imported from /Users/{user}/.kit/api/kit.js
    at new NodeError (node:internal/errors:371:5)
    at finalizeResolution (node:internal/modules/esm/resolve:418:11)
    at moduleResolve (node:internal/modules/esm/resolve:983:10)
    at defaultResolve (node:internal/modules/esm/resolve:1080:11)
    at ESMLoader.resolve (node:internal/modules/esm/loader:530:30)
    at ESMLoader.getModuleJob (node:internal/modules/esm/loader:251:18)
    at ESMLoader.import (node:internal/modules/esm/loader:332:22)
    at importModuleDynamically (node:internal/modules/esm/translators:106:35)
    at importModuleDynamicallyCallback (node:internal/process/esm_loader:35:14)
    at global.attemptImport (file:///Users/liljack/.kit/api/kit.js:140:30)
```

I even tried to create a new script or directly reinstalling all the kit software but the problem persists.
I can't understand what I'm doing wrong.

Thank you in advance !.

**Comment by johnlindquist** [comment]
Try quitting and deleting the ~/.kit dir then open the app again.

The most common issue is that the ~/.kit/node_modules encounters an install error when it updates. I'm trying to find better ways to surface that to the users.

**Reply by jacopo-degattis** [reply]
Perfect, I'll try, thank you very much !
Have a nice day :)

**Reply by johnlindquist** [reply]
🙏 Please let me know if this fixes it. It usually does, but I definitely need to know if there's a different issue hiding somewhere.

Also, I can't remember if I sent you an invite to our Discord yet 😇

https://discord.gg/ykxTKSuQ

**Reply by jacopo-degattis** [reply]
Hi, yes it does but the issue keep coming back after some time.
I'll update you whenever the issue reoccurs (hopefully not).
Anyway I'm already in the discord channel, thanks for the invitation !
Thank you again

**Reply by jacopo-degattis** [reply]
> 🙏 Please let me know if this fixes it. It usually does, but I definitely need to know if there's a different issue hiding somewhere.
> 
> Also, I can't remember if I sent you an invite to our Discord yet 😇
> 
> https://discord.gg/ykxTKSuQ

It just happened again, don't know what the problem could be.

**Reply by johnlindquist** [reply]
Does your ~/.kenv/package.json have the ~/.kit dir installed like this?

```json
{
  "type": "module",
  "dependencies": {
    "@johnlindquist/kit": "file:../.kit"
  }
}
```

**Reply by johnlindquist** [reply]
Also, do you have multiple accounts on the same computer?

Kit.app looks for .kit/.kenv/.knode in the user's home directory and only sets it up on the account where you install it.

**Reply by jacopo-degattis** [reply]
Yeah my file look exactly like this, I just have 2 libraries installed that show up in there.
Also I just have one account on my computer right now.

**Reply by johnlindquist** [reply]
What does the KIT var look like in your ~/.kenv/.env?

```properties
KIT=/Users/johnlindquist/.kit
KENV=/Users/johnlindquist/.kenv
KNODE=/Users/johnlindquist/.knode
KIT_TEMPLATE=default
KIT_EDITOR=kit
```

The {user} part of your error is really confusing to me. Somewhere the path where .kit is located is getting messed up...

**Reply by jacopo-degattis** [reply]
It was me, I replaced my username with {user}, sorry for that XD
Anyway my file looks exactly like yours but I also have `KIT_MODE=ts`

**Reply by johnlindquist** [reply]
In the terminal, try:

```bash
cd ~/.kit
PATH=~/.knode/bin:$PATH
npm i
```

This will manually run the npm install that should happen during setup.

Do you see an errors?

**Reply by jacopo-degattis** [reply]
I see this warning

This is my `npm i ` output
```
npm WARN tarball tarball data for strip-ansi@https://registry.npmjs.org/strip-ansi/-/strip-ansi-7.0.1.tgz (sha512-cXNxvT8dFNRVfhVME3JAe98mkXDYN2O1l7jmcwMnOslDeESg1rF/OZMtK0nRAhiari1unG5cD4jG3rapUAkLbw==) seems to be corrupted. Trying again.
npm WARN tarball tarball data for mime@https://registry.npmjs.org/mime/-/mime-3.0.0.tgz (sha512-jSCU7/VB1loIWBZe14aEYHU/+1UMEHoaO7qxCOVJOw9GgH72VAWppxNcjU+x9a2k3GSIBXNKxXQFqRvvZ7vr3A==) seems to be corrupted. Trying again.
npm WARN tarball tarball data for import-from@https://registry.npmjs.org/import-from/-/import-from-4.0.0.tgz (sha512-P9J71vT5nLlDeV8FHs5nNxaLbrpfAV5cF5srvbZfpwpcJoM/xZR3hiv+q+SAnuSmuGbXMWud063iIMx/V/EWZQ==) seems to be corrupted. Trying again.
npm WARN tarball tarball data for ansi-regex@https://registry.npmjs.org/ansi-regex/-/ansi-regex-6.0.1.tgz (sha512-n5M855fKb2SsfMIiFFoVrABHJC8QtHwVx+mHWP3QcEqBHYienj5dHSgjbxtC0WEZXYt4wcD6zrQElDPhFuZgfA==) seems to be corrupted. Trying again.
npm WARN tarball tarball data for emoji-regex@https://registry.npmjs.org/emoji-regex/-/emoji-regex-9.2.2.tgz (sha512-L18DaJsXSUk2+42pv8mLs5jJT2hqFkFE4j21wOmgbUqsZ2hL72NsUU785g9RXgo3s0ZNgVl42TiHp3ZtOv/Vyg==) seems to be corrupted. Trying again.

added 369 packages, removed 11 packages, changed 27 packages, and audited 1041 packages in 3s

119 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities
```

**Reply by johnlindquist** [reply]
Does running a script work now? Or same error?

**Reply by johnlindquist** [reply]
Also, do you see the

`~/.kenv/.scripts` dir? (Notice the "." before scripts.)

Does the content of those scripts look like a JS version of your TS scripts?

**Reply by jacopo-degattis** [reply]
If I run old scripts it does work but if I try to create a new one and execute it it gives me an error.
It doesn't create the .mjs file in the .scripts folder.

Here the error
```
Cannot find module '/Users/liljack/.kenv/.scripts/prova.mjs' imported from /Users/liljack/.kit/api/kit.js
Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/Users/liljack/.kenv/.scripts/prova.mjs' imported from /Users/liljack/.kit/api/kit.js
    at new NodeError (node:internal/errors:371:5)
    at finalizeResolution (node:internal/modules/esm/resolve:418:11)
    at moduleResolve (node:internal/modules/esm/resolve:983:10)
    at defaultResolve (node:internal/modules/esm/resolve:1080:11)
    at ESMLoader.resolve (node:internal/modules/esm/loader:530:30)
    at ESMLoader.getModuleJob (node:internal/modules/esm/loader:251:18)
    at ESMLoader.import (node:internal/modules/esm/loader:332:22)
    at importModuleDynamically (node:internal/modules/esm/translators:106:35)
    at importModuleDynamicallyCallback (node:internal/process/esm_loader:35:14)
    at global.attemptImport (file:///Users/liljack/.kit/api/kit.js:140:30)
`
``

**Reply by johnlindquist** [reply]
1. Did running old scripts always work or did running `npm i` fix it?

2. Are you on latest 1.30.22?

3. What editor are you using to create the script?

4. In your ~/.kit/logs/kit.log, do you see a "Build" log line that looks like it's made by this?
https://github.com/johnlindquist/kitapp/blob/main/src/build.ts#L9

**Reply by jacopo-degattis** [reply]
1. Running old scripts always work but if I update the code it doesn't affect the result. It's like if it never gets compiled again.
2. Yes I'm on the latest 1.30.22
3. Right now I'm using Vscode / Vscodium to create scripts.
4. Yes I see lot of them.

Anyway now it started working again, it's kinda random to be honest.
I can't see the pattern here...
Thanks anyway for all the support

**Reply by johnlindquist** [reply]
Ah, ok. It must be the way the watcher triggers a script to re-build the file. I can put together a better/more consistent approach
